Ginger/Ginger/ALM/Repository/OctaneRepository.cs

```diff
@@ -55,7 +109,117 @@ public override void ExportBfActivitiesGroupsToALM(BusinessFlow businessFlow, Ob
+            {
+                matchingTS = ((OctaneCore)ALMIntegration.Instance.AlmCore).GetTestSuiteById(businessFlow.ExternalID);
+                if (matchingTS != null)
+                {
+                    //ask user if want to continute
+                    userSelec = Reporter.ToUser(eUserMsgKey.BusinessFlowAlreadyMappedToTC, businessFlow.Name, matchingTS.Name);
+                    if (userSelec == Amdocs.Ginger.Common.eUserMsgSelection.Cancel)
+                    {
+                        return false;
+                    }
+                    else if (userSelec == Amdocs.Ginger.Common.eUserMsgSelection.No)
+                    {
+                        matchingTS = null;
+                        testPlanUploadPath = SelectALMTestPlanPath();
+                        if (String.IsNullOrEmpty(testPlanUploadPath))
+                        {
+                            //no path to upload to
+                            return false;
+                        }
+                        //create upload path if checked to create separete folder
+                        if (QCTestPlanFolderTreeItem.IsCreateBusinessFlowFolder)
+                        {
+                            try
+                            {
+                                string folderId = octaneCore.GetLastTestPlanIdFromPath(testPlanUploadPath).ToString();
+                                folderId = octaneCore.CreateApplicationModule(businessFlow.Name, businessFlow.Description, folderId);
+                                testPlanUploadPath = folderId;
+                            }
+                            catch (Exception ex)
+                            {
+                                Reporter.ToLog(eLogLevel.ERROR, "Failed to get create folder for Test Plan with Octane REST API", ex);
+                            }
+                        }
+                        else
+                        {
+                            testPlanUploadPath = octaneCore.GetLastTestPlanIdFromPath(testPlanUploadPath).ToString();
+                        }
+                    }
+                    else
+                    {
+                        if (String.IsNullOrEmpty(testPlanUploadPath))
+                        {
+                            testPlanUploadPath = matchingTS.ParentId;
+                        }
+                    }
+                }
+            }
+
+            testLabUploadPath = testPlanUploadPath;
+            //check if all of the business flow activities groups already exported to Octane and export the ones which not
+            foreach (ActivitiesGroup ag in businessFlow.ActivitiesGroups)
+            {
+                if (matchingTS == null)
+                {
+                    matchingTC = null;
+                }
+                else
+                {
+                    matchingTC = new QCTestCase(); 
+                }
+                ExportActivitiesGroupToALM(ag, testPlanUploadPath, false, businessFlow);
```

Comment:
Codacy found an issue: [Remove this default value assigned to parameter 'performSaveAfterExport'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)